app/Models/Contact/Contact.php

```diff
@@ -937,14 +938,50 @@ public function getAvatarURL($size = 110)
             return $this->avatar_external_url;
         }
 
-        $original_avatar_url = Storage::disk($this->avatar_location)->url($this->avatar_file_name);
+        $original_avatar_url = $this->avatar_file_name;
         $avatar_filename = pathinfo($original_avatar_url, PATHINFO_FILENAME);
         $avatar_extension = pathinfo($original_avatar_url, PATHINFO_EXTENSION);
         $resized_avatar = 'avatars/'.$avatar_filename.'_'.$size.'.'.$avatar_extension;
 
         return asset(Storage::disk($this->avatar_location)->url($resized_avatar));
     }
 
+    /**
+     * Delete avatars files.
+     */
+    public function deleteAvatars()
```

Comment:
I don't understand why we need to create a function in Contact object just for this, especially if this process will only happen once for this migration. Is it really useful? Can we move this function elsewhere ?

Reply:
When a contact already has an avatar (not gravatar), if you update the image, today the old image stays in the store forever. I'm not sure it's a voluntary behavior. I've created this function to remove the old image, but we can imagine an other behavior...

Reply:
And today a user can flood us by uploading as much as images he can, and they will stay in the store...

Reply:
This is true. Thanks for the explanation.
I wonder if we should create a service for this method instead, like what I've been started to do here: https://github.com/monicahq/monica/pull/1629/files#diff-b3c40f0913df027e06af970085f889ed

The idea: have a class called, for instance, `deleteAvatarForContact`. This is [explained here](https://medium.com/@remi_collin/keeping-your-laravel-applications-dry-with-single-action-classes-6a950ec54d1d).

What do you think?

Reply:
After a bit reflexion: this does not apply here, even I think Services are great, but it's used for Controller, to handle request the same way for htpp and api. In this case, I already have the $contact object, and I just pass it to the Service: there is no interest.